azavea/raster-foundry
# Add MapToken database migrations, api endpoints

## Overview

This commit adds endpoints to manage map tokens. After this commit
tokens can be created/updated/deleted via API endpoints and should open
up the possibility for wiring up some of the UI components.

### Checklist

~- [ ] Styleguide updated, if necessary~

### Demo

![create](https://cloud.githubusercontent.com/assets/898060/23857191/c2f6fda8-07d2-11e7-988a-91ff2d7bf320.png)
![list-view](https://cloud.githubusercontent.com/assets/898060/23857192/c2f7d818-07d2-11e7-8517-8547179c8694.png)

## Testing Instructions

* Get the project ID to create a map token for and make a `POST` request
with the following data (filling in the fields) to `/api/map-tokens/`

```
{
	"name": "<NAME>",
	"project": "<PROJECT ID>",
	"organizationId": "<ORGANIZATION ID>"
}
```

* Make a `GET` request from `/api/tokens/` - verify that that map token is returned. Make `GET` requests with filters and they work, you may have to create additional projects.

* Make a `PUT` request to update the name of a map token, verify that the name is updated via a `GET` request

Closes #1203 and #1204
